samples/boot/minimal/src/test/java/sample/JwkSetEndpointFilterTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sample;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.http.HttpMethod.GET;
+import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static sample.JwkSetEndpointFilter.WELL_KNOWN_JWK_URIS;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jose.jwk.KeyUse;
+import com.nimbusds.jose.jwk.gen.RSAKeyGenerator;
+
+@TestInstance(Lifecycle.PER_CLASS)
+public class JwkSetEndpointFilterTest {
+
+	private MockMvc mvc;
+	private JWKSet jwkSet;
+	private JWK jwk;
+	private JwkSetEndpointFilter filter;
+
+	@BeforeAll
+	void setup() throws JOSEException {
+		this.jwk = new RSAKeyGenerator(2048).keyID("endpoint-test").keyUse(KeyUse.ENCRYPTION).generate();
+		this.jwkSet = new JWKSet(jwk);
+		this.filter = new JwkSetEndpointFilter(jwkSet);
+		this.mvc = MockMvcBuilders.standaloneSetup(new FakeController()).addFilters(filter)
+				.alwaysDo(print()).build();
+	}
+
+	@Test
+	void testRequestMatcher() {
+		assertTrue(this.filter.ifRequestMatches(new MockHttpServletRequest(GET.name(), WELL_KNOWN_JWK_URIS)));
```

Comment:
Can you change all the assertion checks to use AssertJ instead as that is the primary test lib we use in Spring Security.